### PR TITLE
Feature/yqdrw qe s 11 update form page

### PR DESCRIPTION
**Pull request by an-nguyen-jh**

### CODE REVIEW

File: src/apiService/customClass.js [hunk 1]
```diff
@@ -14,33 +14,40 @@ export class DeviceInfo {
...
       createdTime: deviceInfo.createdTime,
+      computer: {
+        ...deviceInfo.computer,
+      },
+      mouse: {
+        ...deviceInfo.mouse,
+      },
+      screen: {
+        ...deviceInfo.screen,
+      },
```
Comment (anh-dinh-jh) [inline src/apiService/customClass.js:52]: What is the purpose of using spread syntax?

Comment (an-nguyen-jh) [reply src/apiService/customClass.js:52]: It makes copies for all properties of an object to another. So you don't need to write each property and assign it

Comment (anh-dinh-jh) [reply src/apiService/customClass.js:52]: What's the difference between using spread syntax and assign `mouse: deviceInfo.mouse` directly?

Also from that point, can we just return `deviceInfo` instead in this function?

Comment (an-nguyen-jh) [reply src/apiService/customClass.js:52]: In my opinion, I want to make a deep copy of deviceInfo which doesn't have any relative to the old device info. If return deviceInfo, computer, mouse, screen properties will become a shallow copy and have reference to old deviceInfo

Comment (anh-dinh-jh) [reply src/apiService/customClass.js:52]: So how about using spread syntax on the `deviceInfo`? Such as ` return { ...deviceInfo }`

Comment (an-nguyen-jh) [reply src/apiService/customClass.js:52]: spread operator deep copies the data if it is not nested and `deviceInfo, computer, mouse, screen` is nested. We can use it with deviceRequest` below

**Review by anh-dinh-jh — APPROVED** [review]

**Pull request merged**